### PR TITLE
imagebuilder: add function to show manifest

### PR DESCRIPTION
**Pull request by aparcar**
Tested with 18.06.0-rc2/ar71xx/generic/tl-wdr4300-v1, image & list

This PR is based on the work of @fewckert[1] with slight improvements.

Add function `manifest` to show the manifest of the produced image, before
actually building it. The manifest contains an orderd list of package
name and version.

This is usefull to check package dependencies but also determine a
unique and reproducible image name before building the package. The
sysupgrade server[2] builds images on request with individual package
selection. To distignish between created images which contain differnt
packages, the EXTRA_IMAGE_NAME is set to a shortend hash of the
manifest's content. So far the image was renamed afterwards as the
manifests content was unknown, however this corrupts the signed
sha256sums. This patch allows a clean solution as to dtermine the
manifest in advance and set the EXTRA_IMAGE_NAME accordingly.

[1]: https://github.com/lede-project/source/pull/1591
[2]: https://github.com/aparcar/attendedsysupgrade-server

Signed-off-by: Paul Spooren <mail@aparcar.org>

### CODE REVIEW

**Comment by aparcar** [comment]
@dangowrt & @jow- please have a look this function would be very handy for me

**Comment by aparcar** [comment]
@feckert should this be called `list` or rather `manifest`?

File: target/imagebuilder/files/Makefile [hunk 1]
```diff
@@ -45,6 +45,14 @@ Building images:
 	make image FILES="<path>" # include extra files from <path>
 	make image BIN_DIR="<path>" # alternative output directory for the images
 	make image EXTRA_IMAGE_NAME="<string>" # Add this to the output image filename (sanitized)
+
+Print manifest:
+	List "all" packages which get installed into the image.
```
Comment (dangowrt) [inline target/imagebuilder/files/Makefile]: What is expressed by the quotes around `"all"`? I don't get it.

**Review by dangowrt — CHANGES_REQUESTED** [review]
Please remove the now obsolete paragraph
```
To lower confusion with the `package_list` (which prints *all* available
packages) the function could be renamed to `manifest`.
```
from the patch description.

File: target/imagebuilder/files/Makefile (hunk 1 above)
Comment (aparcar) [reply target/imagebuilder/files/Makefile]: Copied from the original PR, I can remove that. @feckert can you elaborate?

Comment (feckert) [reply target/imagebuilder/files/Makefile]: If you build a new image with imagebuilder and add a package to PACKAGES variable then this package could also have dependency which also get installed. With this sentence i want to explain that the output shows all packages not only the content of the PACKAGES variable.

**Comment by feckert** [comment]
@aparcar I would now prefer manifest

**Comment by blogic** [comment]
merged, Thanks !

**Pull request closed without merging**